modules/packages/BLAS.chpl

```diff
@@ -19,103 +19,142 @@
 -------------------
 
-In order to compile with the BLAS libraries, you will need
-BLAS and CBLAS (C wrappers to BLAS) installed on your system. You
-will also need to know the location of the cblas.h header file,
-as well as the BLAS libraries you need to link in (and where they
-are, if they're not in a standard location).
+In order to compile a Chapel program that uses this module, the
+BLAS and CBLAS (C wrappers to BLAS) libraries must be installed on the system.
+The paths to both the ``cblas.h`` header file and BLAS library
+must be passed to the ``-I`` and ``-L`` compiler arguments. The library name,
+typically ``blas``, must be passed to the ``-l`` argument as well.
 
-We recommend using a version of BLAS optimized for your system
-to get the best performance.
-
-As an example, with OpenBLAS :
+The compilation command should look something like this:
 
 .. code-block:: sh
 
     chpl -I$PATH_TO_CBLAS_DIR \
          -L$PATH_TO_BLAS_LIBS -lblas source.chpl
 
-You may need additional arguments for different libraries. For instance, if you
-use ATLAS, you will likely need to also add a -latlas to the above
-command. We strongly recommend referring to the documentation for the
-version of BLAS you have.
+BLAS Implementations:
+  There is a wide range of
+  `BLAS implementations <https://en.wikipedia.org/wiki/Basic_Linear_Algebra_Subprograms#Implementations>`_
+  available.
+  This module was built and tested with `netlib's CBLAS
+  <http://www.netlib.org/blas/#_cblas>`_, but many other implementations are
+  compatible as well.
+  Using a version of BLAS optimized for the user's system will yield the best
+  performance.
+
+  There are a few known caveats with some popular implementations:
+
+  * **ATLAS**
+
+    * The compilation command above likely requires the additional flag: ``-latlas``
+
+  * **MKL**
+
+    * The BLAS module assumes that the CBLAS functions are defined in the
+      header named ``cblas.h``.
+      MKL defines these in ``mkl_cblas.h``. This can be worked around by
+      creating a symbolic link to the correct header file name:
+      ``ln -s mkl_cblas.h cblas.h``
+
+  * **OpenBLAS**
+
+    * The header files that are included with OpenBLAS differ from the reference
+      CBLAS prototypes for complex arguments by using ``float*`` and ``double*``
+      pointers, instead of ``void*`` pointers.  Using this will likely result in
+      warnings about incompatible pointer types. These may be ignored.
+
+Cray Systems:
+  The **PBLAS** implementation is made available through Cray's libsci, which
+  comes installed on all Cray systems. To compile programs with the ``BLAS``
+  module on Cray systems, load the ``cray-libsci`` module and link against the
+  ``sci_gnu`` library, using the module-defined ``$CRAY_LIBSCI_PREFIX_DIR``
+  environment variable as shown here:
+
+.. code-block:: sh
+
+  module load cray-libsci
+
+  chpl -I$CRAY_LIBSCI_PREFIX_DIR/include
+       -L$CRAY_LIBSCI_PREFIX_DIR/lib -lsci_gnu
```

Comment:
@ben-albrecht -- is this actually necessary? I just tried doing chpl test_blas3.chpl and it compiled just fine (on a Cray) (after doing module load cray-libsci --- or rather after it was loaded by default).  I think the compiler wrappers handle all the rest of the magic.

That should also let one use non-gnu versions as well....